Jenkinsfile

```diff
@@ -0,0 +1,150 @@
+pipeline {
+    agent any
+    environment {
+        container_name = "c_${BUILD_ID}_${JENKINS_NODE_COOKIE}"
+        user_ci = credentials('lsst-io')
+        work_branches = "${GIT_BRANCH} ${CHANGE_BRANCH} develop"
+    }
+    stages {
+        stage("Pulling image.") {
+            steps {
+                script {
+                    sh """
+                    docker pull lsstts/salobj:develop
+                    """
+                }
+            }
+        }
+        stage("Start container") {
+            steps {
+                script {
+                    sh """
+                    chmod -R a+rw \${WORKSPACE}
+                    container=\$(docker run -v \${WORKSPACE}:/home/saluser/repo/ -td --rm --name \${container_name} -e LTD_USERNAME=\${user_ci_USR} -e LTD_PASSWORD=\${user_ci_PSW} lsstts/salobj:develop)
+                    """
+                }
+            }
+        }
+        stage("Checkout sal") {
+            steps {
+                script {
+                    sh """
+                    docker exec -u saluser \${container_name} sh -c \"source ~/.setup.sh && cd /home/saluser/repos/ts_sal && /home/saluser/.checkout_repo.sh \${work_branches} && git pull\"
```

Comment:
I strongly encourage you to break these long lines.

Here is an example: https://github.com/lsst-ts/ts_salobj/blob/develop/Jenkinsfile

Reply:
Also would it make any sense to combine these stages into one long shell script? It would reduce a lot of boilerplate, at the expense of reducing granularity (I'm not sure how bad that would be).

I think it would be reasonably practical with breaking long lines. For example:
```
"source ~.setup.sh && " +
"cd /home/saluser/repos/ts_sal && " +
"/home/saluser/.checkout_repo.sh \${work_branches} && git pull && " +
"cd ../ts_salobj && " +
"/home/saluser/.checkout_repo.sh \${work_branches} && git pull && " +
"cd ../ts_xml && " +
"/home/saluser/.checkout_repo.sh \${work_branches} && git pull && " +
"cd ../ts_idl && " +
"/home/saluser/.checkout_repo.sh \${work_branches} && git pull && " +
"echo Build IDL files  && " +
"setup ts_sal -t current && " +
"make_idl_files.py Rotator && " +
```
though I suspect you would overflow a buffer if you tried to do *everything* in one long string.

Reply:
The multiple sub-stages help debugs problems when they occur, so I'd rather keep them.